learn/best_practices/arrays.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: book
+title: Arrays
+permalink: /learn/best_practices/arrays
+---
+
+When passing arrays in and out of a subroutine/function, use the
+following pattern for 1D arrays (it is called <span
+class="title-ref">assumed-shape</span>):
+
+``` fortran
+subroutine f(r)
+real(dp), intent(out) :: r(:)
+integer :: n, i
+n = size(r)
+do i = 1, n
+    r(i) = 1.0_dp / i**2
+enddo
```

Comment:
```suggestion
end do
```

Reply:
I have just pushed a commit for those three `enddo`:
```bash
$ grep -inRI "enddo"
arrays.md:18:enddo
arrays.md:64:enddo
arrays.md:94:enddo
```